sc/shared/src/test/scala/sigmastate/TestingInterpreterSpecification.scala

```diff
@@ -215,6 +216,46 @@ class TestingInterpreterSpecification extends CompilerTestingCommons
     testWithCasting("toBigInt")
   }
 
+  property("BigInt downcasting to byte") {
+    def test() = testEval("{ sigmaProp(0L.toBigInt.toByte <= CONTEXT.preHeader.version) }")
+    if(VersionContext.current.isV6SoftForkActivated) {
+      test()
+    } else {
+      an[Exception] shouldBe thrownBy(test())
+    }
+  }
+
+  property("BigInt downcasting to short") {
+    def test() = testEval("{ sigmaProp(0L.toBigInt.toShort <= CONTEXT.preHeader.version.toShort) }")
+    if(VersionContext.current.isV6SoftForkActivated) {
+      test()
+    } else {
+      an[Exception] shouldBe thrownBy(test())
+    }
+  }
+
+  property("BigInt downcasting to int") {
+    def test() = testEval("{ sigmaProp(1L.toBigInt.toInt < CONTEXT.preHeader.timestamp.toInt) }")
+    if(VersionContext.current.isV6SoftForkActivated) {
+      test()
+    } else {
+      an[Exception] shouldBe thrownBy(test())
+    }
+  }
+
+  property("BigInt downcasting to long") {
+    def test() = testEval("{ sigmaProp(1L.toBigInt.toLong < CONTEXT.preHeader.timestamp) }")
+    if(VersionContext.current.isV6SoftForkActivated) {
+      test()
+    } else {
+      an[Exception] shouldBe thrownBy(test())
+    }
+  }
+
+  property("upcasting to bigint") {
+    testEval("{ sigmaProp(1L.toBigInt < bigInt(\"2\")) }")
+  }
+
```

Comment:
Every v6.0 change should be tested in LanguageSpecificationV6, this is the only way to ensure:
1) existing behavior is preserved for old versions (including required failures)
2) the new behavior is enabled starting from the given version
2) the implementation is complete, i.e. all the compiler and interpreter components work as expected
In this specific case, changeFeature can be used.

Separate unit tests cannot ensure test coverage.

Reply:
this test is non-versioned, so nothing to bring to LSV6 from it

For versioned tests, alternatives are added to LSV6.